apps/src/templates/instructions/TopInstructions.jsx

```diff
@@ -725,9 +739,18 @@ class TopInstructions extends Component {
             )}
             {isCSF &&
               this.props.viewAs === ViewType.Teacher &&
-              this.props.teacherMarkdown &&
-              this.state.tabSelected === TabType.TEACHER_ONLY && (
-                <TeacherOnlyMarkdown ref="teacherOnlyTab" />
+              (this.props.hasContainedLevels ||
+                (this.props.teacherMarkdown &&
+                  this.state.tabSelected === TabType.TEACHER_ONLY)) && (
```

Comment:
you are on the right track here, but it looks like you might need to remove this line from the conditional so that the solution does not get unrendered and lost forever when you switch to a different tab.